### PR TITLE
doc/nixos: Explain better NixOS specific Systemd stuff

### PR DESCRIPTION
**Pull request by doronbehar**
Divide the "Service Management" chapter into two sections. The 1st (the
original) explaining General, not NixOS specific ways to interact with
Systemd. The 2nd section, explaining NixOS specific things worth
knowing.

Explain in the 2nd section a bit NixOS modules and services of Nixpkgs,
and mention `systemd.user.services` option. Give an example
demonstrating how to enable imperatively an upstream provided unit file
for a user. Explain why `systemctl --user enable` doesn't work for the
long term on NixOS.

###### Motivation for this change

My own confusion regarding the subject + https://discourse.nixos.org/t/how-to-enable-upstream-systemd-user-services-declaratively/7649/5 .

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/administration/service-mgmt.xml
Comment (andir) [inline nixos/doc/manual/administration/service-mgmt.xml:104]: Do you really have to create that dir manually or isn't that done by the `systemctl --user enable …` below?

Comment (doronbehar) [reply nixos/doc/manual/administration/service-mgmt.xml:104]: It may not exist if it's a first time when you use systemd for anything. The point of this `<screen>`, is to enable a Systemd user service **without** `systemctl --user enable`.

Comment (rnhmjoj) [reply nixos/doc/manual/administration/service-mgmt.xml:104]: From experience, I can confirm this is the correct way to enable user services. 
Using `systemctl --user enable` will create a link that will break once the package hash changes and the directory may not exist.

File: nixos/doc/manual/administration/service-mgmt.xml
Comment (rnhmjoj) [inline nixos/doc/manual/administration/service-mgmt.xml:90]: I don't agree with this statement. We generally prefer to *use* the upstream unit, especially when they contains complex settings like for security hardening. This whould just put more burder on the maintainer and reduce security overall.

Also, pulling in an upstream unit doesn't mean we just hand over control of the service: units can be overriden just fine from the declarative systemd configuration. There are [several]([https://github.com/NixOS/nixpkgs/search?q=use+upstream+unit&type=commits) NixOS module doing this.

Comment (doronbehar) [reply nixos/doc/manual/administration/service-mgmt.xml:90]: You are correct. What confused me is that `systemd.packages` enables and starts system service, but not user services on boot. I've documented this after thorough testings.

**Review by rnhmjoj — CHANGES_REQUESTED** [review]
Nit: systemd is [spelled](https://www.freedesktop.org/wiki/Software/systemd/#spelling) in lower case.

**Comment by doronbehar** [comment]
> Nit: systemd is [spelled](https://www.freedesktop.org/wiki/Software/systemd/#spelling) in lower case.

Correct. I've addressed that also in the parts I took as is.

Let's wait a bit on this PR, and see if anyone from https://github.com/NixOS/nixpkgs/issues/80038 will have something to comment / suggest to add.

**Comment by doronbehar** [comment]
I don't think there's anything more important to add and I think it's better to have this then nothing. Fixed the systemd spelling.

File: nixos/doc/manual/administration/service-mgmt.xml
Comment (rnhmjoj) [inline nixos/doc/manual/administration/service-mgmt.xml:100]:
```suggestion
   <literal>services.#name#.enable = true;</literal>. These services are defined
```

**Review by rnhmjoj — CHANGES_REQUESTED** [review]
I agree, just fix the build error and I would merge.

**Comment by doronbehar** [comment]
> I agree, just fix the build error and I would merge.

Great. I fixed the typo and made sure it builds. Will wait for your final approval.

**Review by rnhmjoj — APPROVED** [review]
Thank you! Contribution to the docs are always greatly appreciated.

**Pull request merged**
